### ISSUE TITLE
generate javascript with same line numbers as coffeescript.

### ISSUE THREAD

**Issue by JasonWoof** [jashkenas/coffeescript#3313]
I'd like a mode (commandline flag or whatever) so that the generated javascript has the same line numbers as the coffeescript file.

This would be useful for development, because error messages would have the correct line number.

Here's my test.coffee:

```
a = 'A'
b = a.b.c
```

When I `coffee test.coffee` I get:

```
TypeError: Cannot read property 'c' of undefined
    at Object.<anonymous> (/path/to/test.coffee:6:10)
```

Nope, that's not line 6, that's line 2. Presumably, the coffeescript mode for node.js is converting to javascript automatically. I'd like this automatic converting to preserve line numbers. (And I'd like to have the option of this type of output with `coffee -c` also.)

Here's what I get from `coffee -c test.coffee`:

```
// Generated by CoffeeScript 1.4.0
(function() {
  var a, b;

  a = 'A';

  b = a.b.c;

}).call(this);
```

I'd like a mode where the coffee compiler would generate this from my test.coffee:

```
(function() { var a, b; a = 'A';
  b = a.b.c;
}).call(this);
```

that way I'll get the right line number in error messages. This will save me a bunch of time debugging, because I can jump (in my coffeescript file) to the line number in the error message, instead of searching/guessing around trying to find some variable/etc name mentioned in the error message. A few times I've had to compile a source file to coffeescript to look at what's happening at that line number in javascript before I could figure out what line in my coffeescript was causing the error.

**Comment by adamziel** [comment]
+1 for preserving line numbers. Source maps are not suitable for every use case. In particular they are pretty slow when working with a big codebase.

**Comment by toomim** [comment]
@adamziel Good insight. I just ran into this too.

I'm writing live-editing code from CoffeeScript. With every keystroke in the programmer's editor, it recompiles the file and loads the new code into the browser. Source maps are very large, and made this slower... I'll try to take some measurements.

**Comment by danschumann** [comment]
Even with a big codebase, there is gulp-cache, gulp-remember, and of course
gulp-sourcemaps.   I would want this feature, if I wasn't using source
maps.  If you're going to fix a problem like this, fix it all the way and
just sourcemap it.

I would be for it, but I don't want to do it, and I've seen better
suggestions get turned down, like better multiline arguments for functions
and deconstructing rules, for mere lack of people willing to hack on it.

**Comment by adamziel** [comment]
gulp-* packages are great to eliminate build-time delays, however there's a significant delay in a browser too. For a well-sized codebase it takes up to 30 seconds for chrome to load and process source maps **on every refresh**. I totally understand that it's probably not a problem for most small and medium sized applications.

**Comment by toomim** [comment]
Gulp also doesn't help with the live editing situation.

**Comment by akre54** [comment]
Try using webpack for this. You can use [a devtool](http://webpack.github.io/docs/configuration.html#devtool) to speed things up significantly in development.

**Comment by danschumann** [comment]
@adamziel maybe it's because you're `concat`ing everything in dev mode, so it invalidates the entire codebase?  I have a fairly sizable app and it doesn't take nearly that long.  How many lines we talking, 200,000?

**Comment by adamziel** [comment]
@danschumann more than 600,000; besides the initial in-browser load time there was a big slowdown in browser's speed; mostly when debugging long tracebacks that spanned multiple files. The closest thing I got to smooth experience was webpack with it's great dev server (mentioned above by @akre54) however it was unsuitable for other reasons. I ended up using [jsx](https://facebook.github.io/jsx/) with --harmony option - it preserves line numbers and gives some perks of ES6 syntax.
